packages/miniflare/src/plugins/core/index.ts

```diff
@@ -231,16 +235,20 @@ function getCustomServiceDesignator(
 	workerIndex: number,
 	kind: CustomServiceKind,
 	name: string,
-	service: z.infer<typeof ServiceDesignatorSchema>
+	service: z.infer<typeof ServiceDesignatorSchema>,
+	// override with router worker
+	hasAssetsAndIsVitest: boolean = false
 ): ServiceDesignator {
 	let serviceName: string;
 	let entrypoint: string | undefined;
 	if (typeof service === "function") {
 		// Custom `fetch` function
 		serviceName = getCustomServiceName(workerIndex, kind, name);
 	} else if (typeof service === "object") {
+		// Worker with entrypoint
 		if ("name" in service) {
 			if (service.name === kCurrentWorker) {
+				// self binding e.g. for Vitest
 				serviceName = getUserServiceName(refererName);
```

Comment:
This case should also cover the router worker switch

Reply:
Might've misunderstood something, but this branch is for RPC entrypoints right, which _shouldn't_ get assets...? To reflect prod behaviour.

Reply:
Oh fair point, I had thought that you could define things as `{ name: "service-name" }` with entrypoint being optional—didn't realise this was just for RPC

Reply:
I think what you mean is that a `fetch()` handler on a `WorkerEntryPoint` type entry point (whether it is the default or a named entry-point) will check assets first (i.e. go via the Router Worker).

But actual RPC invocations (not to `fetch()`) such as `await env.WORKER.foo()` will not check assets first and will go directly to the User Worker.

Right?

Reply:
Also other non-`fetch()` handlers such as `scheduled` and `queue` will not go via Router Worker

Reply:
Yes sorry that was imprecise - I meant fetch on a worker entrypoint should return assets. Although currently SELF can only hit the default entrypoint (with fetch or RPC) right? So you still couldn't replicate prod behaviour for assets + fetch on a named entrypoints, since we decided importing the worker directly won't return assets.

Reply:
Can't you have service bindings to Worker + Assets in Vitest tests?

Reply:
Right. Sounds right.